sdk/src/main/java/io/radar/sdk/RadarApiClient.kt

```diff
@@ -116,83 +131,83 @@ internal class RadarApiClient(
         val options = RadarSettings.getTrackingOptions(context)
         val tripOptions = RadarSettings.getTripOptions(context)
         try {
-            params.putOpt("id", RadarSettings.getId(context))
-            params.putOpt("installId", RadarSettings.getInstallId(context))
-            params.putOpt("userId", RadarSettings.getUserId(context))
-            params.putOpt("deviceId", RadarUtils.getDeviceId(context))
-            params.putOpt("description", RadarSettings.getDescription(context))
-            params.putOpt("metadata", RadarSettings.getMetadata(context))
+            params.putValid("id", RadarSettings.getId(context))
+            params.putValid("installId", RadarSettings.getInstallId(context))
+            params.putValid("userId", RadarSettings.getUserId(context))
+            params.putValid("deviceId", RadarUtils.getDeviceId(context))
+            params.putValid("description", RadarSettings.getDescription(context))
+            params.putValid("metadata", RadarSettings.getMetadata(context))
```

Comment:
If `putValid()` is only validating `Double`s, does it need to be called for other types?

Reply:
This fix is based off the source of the JSONObject code, which checks if the `value` is a `Number` and if so, performs the same `toDouble()` and infinite/NaN checks, which is the root of the failure for the stack trace posted on the ticket.